### ISSUE TITLE
[dev-server-storybook] Bug on Windows: storybook files are outside the root directory (slash and backslash issue)

### ISSUE THREAD

**Issue by jfernandezpe** [modernweb-dev/web#1229]
Hi all,

I've just created a new componen with `npm init @open-wc` and I'm not able to run storybook. I get an error that says that the storybook files are outside the root directory (log below). Apparenly is a problem with the slash (`/`) and backslash (`\`) chars.

**Environment**: 
node version is 14.8.0.
OS is Windows 10.

**Steps**:

**1 -** Created the componet with the open-wc's tool and do not modify any file

**2 -** Execute  `npm run storybook`, which is `web-dev-server -c .storybook/server.mjs`, 

This should get the next error
``` 
Web Dev Server started...

  Root dir: C:\Users\jfernandezpe\workspace\test-storybook\story-book
  Local:    http://localhost:8000/
  Network:  http://192.168.1.131:8000/


  Error: [@web/dev-server-storybook] The file C:/Users/jfernandezpe/workspace/test-storybook/story-book/stories/index.stories.js is not accessible by the browser because it is outside the root directory C:\Users\jfernandezpe\workspace\test-storybook\story-book. Change the rootDir option to include this directory.      at createError (C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-storybook\dist\shared\utils.js:9:12)
      at Object.createBrowserImport (C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-storybook\dist\shared\utils.js:14:15)
      at C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-storybook\dist\shared\stories\findStories.js:16:70
      at Array.map (<anonymous>)
      at Object.findStories (C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-storybook\dist\shared\stories\findStories.js:16:41)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
      at async Object.serve (C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-storybook\dist\serve\storybookPlugin.js:59:53)
      at async C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-core\dist\middleware\pluginServeMiddleware.js:20:30    
      at async C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-core\dist\middleware\pluginMimeTypeMiddleware.js:15:9  
      at async C:\Users\jfernandezpe\workspace\test-storybook\story-book\node_modules\@web\dev-server-core\dist\middleware\pluginTransformMiddleware.js:34:9 
```

I've been debugging a little bit and I think that here is the problem: 

https://github.com/modernweb-dev/web/blob/master/packages/dev-server-storybook/src/shared/utils.ts#L8
This line is matching the rootDir in linux blackslash syntax (`\`) and filePath is in the windows slash syntax (`/`), like the log.

Best regards.

**Comment by kishtner** [comment]
I have the same issue. The line @jfernandezpe pointed out seems to be the problem. I changed the condition in line 13 to not trip on Windows paths and its working for now. Thanks for the tip, @jfernandezpe .

**Comment by LarsDenBakker** [comment]
Thanks for reporting, can you try if @web/dev-server-storybook@0.3.4 solves the issue?

**Comment by kishtner** [comment]
Unfortunately, the exception is still thrown. I debugged it; these are the values for the paths I get in `\shared\utiltils.js:10:12`

`filePath = 'C:/Users/xyz/_dev/workspace/research-assistant/out-tsc/stories/research-assistant.stories.js'`
`rootDir = 'C:\\Users\\xyz\\_dev\\workspace\\research-assistant'`

Seems the `\\` are the problem, since they do not seem to be removed even in the final check in `path-is-inside`.

I hope this helps. If you need anything else, let me know.

**Comment by kishtner** [comment]
Quick follow up:
Everything seems to be ok until the final check: `thePath[potentialParent.length] === path.sep`. 
This fails, i guess, because `path.sep` is `\\` and `thePath[potentialParent.length]` is `/`

**Comment by LarsDenBakker** [comment]
Thanks for digging into this further... I thought by using `path-is-inside` we would avoid these kinds of issues, because in fact a `/` is a valid path separator on windows. I've published @web/dev-server-storybook@0.3.5 now which normalizes the separators on our side... Could you try this version? I'm unable to test this on windows myself right now which isn't great.

**Comment by kishtner** [comment]
Looking good! 
I created a new project using `npm init @open-wc` and `npm run storybook` displays the default story correctly.

Thanks!
